### ISSUE TITLE
Sentinel commands have migrated in redis main 

### ISSUE THREAD

**Issue by sarathak** [jazzband/django-redis#542]
Redis pre release test throwing error in CI

- Test Python 3.9, Django 3.2, Redis.py master
- Test Python 3.9, Django 3.2 main, Redis.py master

```python
  self = Sentinel<sentinels=[127.0.0.1:26379]>, service_name = 'default_service'
  
      def discover_master(self, service_name):
          """
          Asks sentinel servers for the Redis master's address corresponding
          to the service labeled ``service_name``.
      
          Returns a pair (address, port) or raises MasterNotFoundError if no
          master is found.
          """
          for sentinel_no, sentinel in enumerate(self.sentinels):
              try:
  >               masters = sentinel.sentinel_masters()
  E               AttributeError: 'Redis' object has no attribute 'sentinel_masters'
  
  .tox/py39-dj32-redismaster/lib/python3.9/site-packages/redis/sentinel.py:211: AttributeError
  _____ ERROR at teardown of DjangoRedisCacheTestEscapePrefix.test_iter_keys _____
  
  self = <tests.test_backend.DjangoRedisCacheTestEscapePrefix testMethod=test_iter_keys>
  
      def tearDown(self):

```

**Comment by WisdomPill** [comment]
hello @sarathak,
redis master is allowed to fail as there might be new unreleased changes that will break the library.
Have you had look at what this might be?
Are you open to make changes to solve this for newer version of `redis-py`?

**Comment by sarathak** [comment]
@WisdomPill 

my opinion is we have to remove pre release test from CI

**Comment by WisdomPill** [comment]
why one earth would you want to do that?

pre releases is added for this very reason, so that we know that before releasing a new version we have to solve some issues first to support the next version of redis

**Comment by WisdomPill** [comment]
In this case, sentinel and cluster commands have moved to it's own mixins, have a look [here](https://github.com/andymccurdy/redis-py/pull/1534). New commands are found [here](https://github.com/andymccurdy/redis-py/blob/e19a76c58f2a998d86e51c5a2a0f1db37563efce/redis/commands.py#L2953)

**Comment by chayim** [comment]
We're planning on releasing a new redis-py in the short-term. This is a **great catch** that your project has in its pipeline, I highly recommend maintaining these tests. Note: We haven't broken the API itself, but I'd like to go down the path of more restructuring in the mid-term.

**Comment by WisdomPill** [comment]
@chayim thanks for the notice. 
Can you elaborate what kind of restructuring you have in mind? Just so we can get an idea of what and where do the changes. 
Is this new version going to be a major release or not?

**Comment by chayim** [comment]
Absolutely! In the short term - nothing. There's just these mixins. In the
long-term we'd like to get better cluster support, the string argument
building in place, etc. General goal is to *never* break an API.

Originally, I was pushing for the next release to be 3.6, but given that
there are many changes, and that projects have frequently pinned >=3.5.3 &&
< 4.0.0, I think 4.0 and onwards makes sense.

>

**Comment by hartwork** [comment]
This issue is supposedly fixed by commit https://github.com/redis/redis-py/commit/d2b233384458869270352b8c99ca682ae480da5f that is included in redis-py >=4.0.**2** already today. The commit adds method `Redis.sentinel_masters` by making class `Redis` inherit from `SentinelCommands` where the related code resides.

As a result, blocking redis-py 4.0.2 in pull request #570 may cause more harm than good and keeping users in the past with redis-py. Another user of redis-py — the kombu project — only blocks redis-py 4.0.**0** and 4.0.**1** for a week now, see https://github.com/celery/kombu/pull/1445/files . 

What are your thoughts on unblocking redis-py 4.0.**2** in django-redis? 

CC https://github.com/hartwork/jawanndenn/pull/357

**Comment by hartwork** [comment]
PS: I have created a related pull request #576 for review just now.
